src/Npgsql/NpgsqlTypes/NpgsqlRange.cs

```diff
@@ -19,109 +20,176 @@
 // AND FITNESS FOR A PARTICULAR PURPOSE. THE SOFTWARE PROVIDED HEREUNDER IS
 // ON AN "AS IS" BASIS, AND THE NPGSQL DEVELOPMENT TEAM HAS NO OBLIGATIONS
 // TO PROVIDE MAINTENANCE, SUPPORT, UPDATES, ENHANCEMENTS, OR MODIFICATIONS.
+
 #endregion
 
 using System;
+using System.ComponentModel;
+using System.Globalization;
 using System.Text;
 using JetBrains.Annotations;
-
-#pragma warning disable 1591
+using Microsoft.Extensions.Primitives;
 
 // ReSharper disable once CheckNamespace
 namespace NpgsqlTypes
 {
-    public struct NpgsqlRange<T> : IEquatable<NpgsqlRange<T>>
+    /// <summary>
+    /// Represents a PostgreSQL range type.
+    /// </summary>
+    /// <typeparam name="T">The element type of the values in the range.</typeparam>
+    /// <remarks>
+    /// See: https://www.postgresql.org/docs/current/static/rangetypes.html
+    /// </remarks>
+    [TypeConverter(typeof(NpgsqlRange<>.RangeTypeConverter))]
+    public readonly struct NpgsqlRange<T> : IEquatable<NpgsqlRange<T>>
     {
-#pragma warning disable CA1000
+        /// <summary>
+        /// Represents the empty range.
+        /// </summary>
         public static NpgsqlRange<T> Empty { get; } = new NpgsqlRange<T>(default, default, RangeFlags.Empty);
-#pragma warning restore CA1000
 
+        /// <summary>
+        /// The lower bound of the range. Only valid when <see cref="LowerBoundInfinite"/> is false.
+        /// </summary>
+        [CanBeNull]
         public T LowerBound { get; }
+
+        /// <summary>
+        /// The upper bound of the range. Only valid when <see cref="UpperBoundInfinite"/> is false.
+        /// </summary>
+        [CanBeNull]
         public T UpperBound { get; }
 
-        internal RangeFlags Flags { get; set; }
+        /// <summary>
+        /// The characteristics of the boundaries.
+        /// </summary>
+        internal readonly RangeFlags Flags;
 
-        public bool LowerBoundIsInclusive
-        {
-            get => (Flags & RangeFlags.LowerBoundInclusive) != 0;
-            private set
-            {
-                if (value)
-                    Flags |= RangeFlags.LowerBoundInclusive;
-                else
-                    Flags &= ~RangeFlags.LowerBoundInclusive;
-            }
-        }
+        /// <summary>
+        /// True if the lower bound is part of the range (i.e. inclusive); otherwise, false.
+        /// </summary>
+        public bool LowerBoundIsInclusive => Flags.HasFlag(RangeFlags.LowerBoundInclusive);
```

Comment:
Ah, I wasn't thinking about the perf implications for the full framework. I'll revert the `Enum.HasFlag()` checks throughout the file.

Though, if those intrinsics are ever backported, it would be a nice readability improvement to use `Enum.HasFlag()`. I'll add a code comment citing this thread for posterity.